cli/tokens.go

```diff
@@ -85,12 +86,41 @@ func createToken() *cobra.Command {
 	return cmd
 }
 
+// tokenListRow is the type provided to the OutputFormatter.
+type tokenListRow struct {
+	// For JSON format:
+	codersdk.APIKey `table:"-"`
+
+	// For table format:
+	ID        string    `json:"-" table:"id,default_sort"`
+	LastUsed  time.Time `json:"-" table:"last used"`
+	ExpiresAt time.Time `json:"-" table:"expires at"`
+	CreatedAt time.Time `json:"-" table:"created at"`
+	Owner     string    `json:"-" table:"owner"`
+}
+
+func tokenListRowFromToken(token codersdk.APIKey, usersByID map[uuid.UUID]codersdk.User) tokenListRow {
+	user := usersByID[token.UserID]
+
+	return tokenListRow{
+		APIKey:    token,
+		ID:        token.ID,
+		LastUsed:  token.LastUsed,
+		ExpiresAt: token.ExpiresAt,
+		CreatedAt: token.CreatedAt,
+		Owner:     user.Username,
+	}
+}
+
 func listTokens() *cobra.Command {
-	formatter := cliui.NewOutputFormatter(
-		cliui.TableFormat([]codersdk.APIKey{}, nil),
-		cliui.JSONFormat(),
+	var (
+		all           bool
+		displayTokens []tokenListRow
+		formatter     = cliui.NewOutputFormatter(
+			cliui.TableFormat([]tokenListRow{}, nil),
```

Comment:
I want to only show the OWNER column if the `--all` flag is specified. I was hoping to pass in some `defaultColumns` after the command has been run (after the value of `all` is known); however, the formatter is declared prior to this. Is there a way to override the formatter in the `cobra.Command` callback?

Reply:
It's a bit problematic since for us to read the `--all` value, the Cobra command needs to be ready. But before it is ready, the formatter needs to register flags (which e.g. output default values). So there's kind of a dependency cycle. It could be solved in two ways, though.

1. Parse `os.Args()` and look for `-a`/`--all`, if found, pass the columns you want (instead of nil) here. Or vice-versa.
2. Extend the `cliui` package so that the displayed columns can be changed after-the-fact (might be hard to fit in the current design, though)

Reply:
Thank you! Went with 1).